cmd/notation/registry.go

```diff
@@ -55,6 +59,56 @@ func getRepositoryClient(ctx context.Context, opts *SecureFlagOpts, ref registry
 	return notationregistry.NewRepository(remoteRepo), nil
 }
 
+// getSignatureRepositoryForSign returns a registry.Repository for Sign.
+// ociImageManifest denotes the type of manifest used to store signatures during
+// Sign process.
+// Setting ociImageManifest to true means using OCI image manifest.
+// Otherwise, use OCI artifact manifest and requires Referrers API.
+func getSignatureRepositoryForSign(ctx context.Context, opts *SecureFlagOpts, reference string, ociImageManifest bool) (notationregistry.Repository, error) {
+	ref, err := registry.ParseReference(reference)
+	if err != nil {
+		return nil, err
+	}
+
+	// generate notation repository
+	return getRepositoryClientForSign(ctx, opts, ref, ociImageManifest)
+}
+
+func getRepositoryClientForSign(ctx context.Context, opts *SecureFlagOpts, ref registry.Reference, ociImageManifest bool) (notationregistry.Repository, error) {
+	authClient, plainHTTP, err := getAuthClient(ctx, opts, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	remoteRepo := &remote.Repository{
+		Client:    authClient,
+		Reference: ref,
+		PlainHTTP: plainHTTP,
+	}
+	repositoryOpts := notationregistry.RepositoryOptions{
+		OCIImageManifest: ociImageManifest,
+	}
+	// using OCI artifact manifest to store signatures. This requires the
+	// existence of Referrers API for Sign process.
+	if !ociImageManifest {
+		if err := remoteRepo.SetReferrersCapability(true); err != nil {
```

Comment:
Why not let oras-go automatically determine whether use referrers api or not?

Reply:
Because oras-go does not have this capability yet. Notation needs to explicitly ping Referrers API.

Reply:
https://github.com/oras-project/oras-go/blob/main/registry/remote/repository.go#L532
I think oras-go has the capability, right?

Reply:
Currently, oras-go does not have this capability yet. See https://github.com/oras-project/oras-go/issues/423 for more details.